plugins/scaffolder-backend/src/scaffolder/tasks/DatabaseWorkspaceProvider.ts

```diff
@@ -29,7 +29,7 @@ export class DatabaseWorkspaceProvider implements WorkspaceProvider {
     path: string;
     taskId: string;
   }): Promise<void> {
-    this.storage.serializeWorkspace?.(options);
+    await this.storage.serializeWorkspace?.(options);
```

Comment:
@benjdlambert This was a culprit of the wrong behaviour. Now workspace clean up works well without extra transactions.

Reply:
Phew, good to hear.